### PR TITLE
Increase temporal resolution in main diagram from 1h to 10min

### PR DESCRIPTION
**Pull request by DL6ER**
Changes proposed in this pull request:

- Increase temporal resolution in main diagram from 1h to 10min.

Before:
screenshot at 2016-11-09 15-19-08

Proposed:
screenshot at 2016-11-09 15-53-58

I added a new API call in order to keep backwards-compatibility. There is no speed/memory drawback as the functions are very similar, the only difference being in the way they put intervals on the acquired data.

Numerical confirmation that everything works as it should:

api.php?overTimeData

> {"domains_over_time":["854","854","828","862","834","852","840","854","831","843","857","970","1399","1732","1033","391"],"ads_over_time":["2","0","1","0","3","0","1","0","1","0","2","15","82","194","46","9"]}

api.php?overTimeData10mins

> {"domains_over_time":["144","142","146","141","135","146","140","140","143","134","154","143","141","141","134","137","139","136","140","146","138","141","151","146","147","141","134","143","137","132","136","144","137","143","149","143","142","136","135","146","144","137","134","142","139","143","153","143","140","137","143","138","132","141","136","140","135","137","154","141","140","141","138","161","145","132","138","144","137","140","177","234","226","258","312","218","167","218","185","145","175","703","366","158","150","197","189","136","168","193","155","200","168","160","104"],"ads_over_time":["0","1","0","1","0","0","0","0","0","0","0","0","0","1","0","0","0","0","0","0","0","0","0","0","1","1","0","1","0","0","0","0","0","0","0","0","0","1","0","0","0","0","0","0","0","0","0","0","0","1","0","0","0","0","0","0","0","0","0","0","1","1","0","0","0","0","0","0","0","0","0","15","17","22","22","12","1","8","7","5","9","95","71","7","4","15","10","0","6","11","1","8","10","5","4"]}

Let me verify correctness for some of the data shown above

- 00:00 - 00:59
1 h resolution: 854
10min resolution: 144+142+146+141+135+146 = 854

- 02:00 - 02:59
1 h resolution: 828
10min resolution: 141+141+134+137+139+136 = 828

- 08:00 - 08:59
1 h resolution: 831
10min resolution: 140+137+143+138+132+141 = 831

- 13:00 - 13:59
1 h resolution: 1732
10min resolution: 185+145+175+703+366+158 = 1732

@pi-hole/dashboard

### CODE REVIEW

**Comment by PromoFaux** [comment]
Interesting Idea! 

Will take a closer look at it later on

**Comment by PromoFaux** [comment]
Approced

**Comment by PromoFaux** [comment]
Approved*

**Comment by Mcat12** [comment]
It looks like the hover size might need to be reduced, as hovering over one time is showing a different time: http://prntscr.com/d87bfg

**Comment by Mcat12** [comment]
Also, a time in the tooltip should be added for each 10min interval. Currently only the hour has it's time in the tooltip.

**Comment by DL6ER** [comment]
I'll have a look tomorrow, how I can include the finer times in the tooltip. I refrained from doing so as that would populate the x axis as well such that everything gets extremely crowded:

screenshot at 2016-11-16 23-42-20

**Comment by Mcat12** [comment]
Ah, then in that case unless there's a flag for not showing them except on tooltips, you can leave it out.

**Comment by DL6ER** [comment]
I read through the documentation of Chart.js, but it doesn't seem possible.

> The line chart usually requires an array of labels. This labels are shown on the X axis. There must be one label for each data point.

Therefore, I have to give empty strings for the data points in between the full hour labels. Unfortunately, I cannot give a separate array to the tooltip routine. I'm experimenting with hiding some of the x axis values, but so far, this does not work nicely.

**Comment by DL6ER** [comment]
Okay, I have it now. The documentation is not very clear on how you would do it, though.
screenshot at 2016-11-17 10-43-43

File: js/pihole/index.js [hunk 1]
```diff
@@ -59,6 +59,15 @@ $(document).ready(function() {
                 display: false
             },
             scales: {
+                xAxes: [{
+                    type: 'time',
```
Comment (PromoFaux) [inline js/pihole/index.js]: I think codacy-bot might be being a bit pickythere....

**Comment by DL6ER** [comment]
I'll address codacy-bot issues later on. However, it seems like it's checks have passed anyhow?

**Comment by DL6ER** [comment]
@PromoFaux @Mcat12 All of codacy's "issues" are resolved.

**Comment by PromoFaux** [comment]
>All of codacy's "issues" are resolved.

We've only just implemented codacy, so we may have a few kinks to iron out there... But the idea is that we can enforce a consistent code style across the whole project, something we've been guilty of neglecting for a long while!

**Comment by PromoFaux** [comment]
Approved

**Comment by DL6ER** [comment]
@PromoFaux It will cause more commits, I think (especially this ' vs " thing). But in the end, it might give better code. I agree.

**Comment by Mcat12** [comment]
This seems a bit odd... Shouldn't all the datapoints' all queries and ad queries add up to the totals? If you add up each datapoint's queries, it goes way over the total.
Image

**Comment by DL6ER** [comment]
No, the unit is "queries per hour", so each 10min interval has to be multiplied by 6 to have the correct unit.

So, if you add up all data points and divide by 6 (because you have 6 times too many points), you will get the correct number.

**Comment by Mcat12** [comment]
Ok, "Queries per hour" can be changed to "Queries over time", because the graph should reflect how many queries happened between each datapoint. The per hour unit is left over from when the graph only showed each hour. I think this will be a bit more intuitive to the user.

**Comment by DL6ER** [comment]
I agree, tooltip is somewhat misleading here. I'll change this asap.

**Comment by DL6ER** [comment]
I removed the normalization to "per hour"
screenshot at 2016-11-19 22-15-16

File: js/pihole/index.js [hunk 2]
```diff
@@ -154,13 +166,14 @@ function updateSummaryData(runOnce) {
...
         // Add data for each hour that is available
         // remove last data point since it not representative
         data.ads_over_time.splice(-1,1);
-        for (hour in data.ads_over_time) {
-            // Add x-axis label
-            timeLineChart.data.labels.push(hour + ":00");
+        for (var hour in data.ads_over_time) {
+            var d = new Date();
+            d.setHours(Math.floor(hour/6),10*(hour%6),0,0);
+            timeLineChart.data.labels.push(d);
```
Comment (DL6ER) [inline js/pihole/index.js]: Okay, look at this update. I just always takes some time to identify what specific setting would be necessary with Chart.js (needless to mention that many links in their docs actually point to nowhere).

Comment (PromoFaux) [inline js/pihole/index.js]: Thanks a lot! Much appreciated.

Yeah, I found their docs to be a bit difficult to navigate at times.

**Comment by DL6ER** [comment]
How about this solution? It makes more clear what we really show here.
untitled4866841

**Comment by Mcat12** [comment]
That looks great!

**Comment by DL6ER** [comment]
Updated the wording as it has also recently been updated in the summary blocks.

untitled46985

**Comment by PromoFaux** [comment]
Approved

**Comment by Mcat12** [comment]
Is this a side-effect of having the 10min increments?
Image

**Comment by Mcat12** [comment]
Approved

**Comment by PromoFaux** [comment]
Yeah I noticed that, and I would say probably yes!

**Comment by DL6ER** [comment]
@PromoFaux @Mcat12 Indeed, the x axis has to end at a "full" hour with momentum.js. However, we provide here too little data to fill the last hour.

**Pull request merged**